misc/lldb_cruby.py

```diff
@@ -723,37 +711,44 @@ def rb_id2str(debugger, command, result, internal_dict):
         pos = (num % ID_ENTRY_UNIT) * ID_ENTRY_SIZE
         id_str = rb_ary_entry(target, ary, pos, result)
         lldb_inspect(debugger, target, result, id_str)
+# END FUNCTION STYLE DECLS
 
-def rb_rclass_ext(debugger, command, result, internal_dict):
-    if not ('RUBY_Qfalse' in globals()):
-        lldb_init(debugger)
 
-    target = debugger.GetSelectedTarget()
-    process = target.GetProcess()
-    thread = process.GetSelectedThread()
-    frame = thread.GetSelectedFrame()
+load_dir, _ = os.path.split(os.path.realpath(__file__))
+
+for fname in glob.glob(f"{load_dir}/lldb_rb/commands/*_command.py"):
+    _, basename = os.path.split(fname)
+    mname, _ = os.path.splitext(basename)
 
-    uintptr_t = target.FindFirstType("uintptr_t")
-    rclass_t = target.FindFirstType("struct RClass")
-    rclass_ext_t = target.FindFirstType("rb_classext_t")
+    print(f"{basename} - {mname}")
 
-    rclass_addr = target.EvaluateExpression(command).Cast(uintptr_t)
-    rclass_ext_addr = (rclass_addr.GetValueAsUnsigned() + rclass_t.GetByteSize())
-    debugger.HandleCommand("p *(rb_classext_t *)%0#x" % rclass_ext_addr)
+    # print(f"importing {mname} from {fname}")
+    exec(f"import lldb_rb.commands.{mname}")
 
+def custom_command_p(memname, mem):
+    return inspect.isclass(mem) and \
+        memname.endswith("Command") and \
+        not memname.startswith("RbBase") and \
+        callable(getattr(mem, "register_lldb_command", None))
 
 def __lldb_init_module(debugger, internal_dict):
```

Comment:
Can we just iterate through `RbBaseCommand.__subclasses__()` which gives an array of subclasses of `RbBaseCommand`? This guarantees every class has a `register_lldb_command` method.

Reply:
I deliberately didn't do that as I didn't want to require commands to have to subclass `RbBaseCommand`. My thought process was that allowing people to use their own class hierarchies, would encourage smaller, more focussed classes that would be easier to read and compose.

Reply:
Ok. so I just refactored this to be a lot more reliant on `RbBaseClass` and I like it much better after all 👍  Thanks Peter